janome/dic.py

```diff
@@ -425,13 +432,34 @@ def buildipadic(self, user_dict, enc):
                 surfaces.append((surface.encode('utf8'), pack('I', morph_id)))
                 entries[morph_id] = (surface, int(left_id), int(right_id), int(
                     cost), part_of_speech, infl_type, infl_form, base_form, reading, phonetic)
+                # update progress
+                if progress_handler:
+                    progress_handler.on_progress()
+
+        # complete progress for reading CSV
+        if progress_handler:
+            progress_handler.on_complete()
+
         inputs = sorted(surfaces)  # inputs must be sorted.
         assert len(surfaces) == len(entries)
-        processed, fst = create_minimum_transducer(inputs)
+
+        # init progress for create_minimum_transducer
+        if progress_handler:
+            progress_handler.on_start(
+                total=len(inputs),
+                desc="Running create_minimum_transducer")
+
+        processed, fst = create_minimum_transducer(inputs,
+            on_progress=progress_handler.on_progress if progress_handler else None)
+
+        # complete progress for create_minimum_transducer
+        if progress_handler:
+            progress_handler.on_complete()
+
         compiledFST = compileFST(fst)
         return compiledFST, entries
 
-    def buildsimpledic(self, user_dict, enc):
+    def buildsimpledic(self, user_dict, enc, progress_handler):
```

Comment:
Can you add the same stuffs to show progresses for this method too (for completeness / consistency)? Or you might be able to take much more sophisticated implementation strategy, by using closures/decorators... (It'd be possible I believe).

Reply:
I've added same code to another. (I'm sorry, I also think sophisticated way can be taken...) 👉 https://github.com/mocobeta/janome/pull/87/commits/e8e32090314a7e55be3695d208b972308c74d306

Reply:
It's fine, thanks. Let me try to integrate two build methods and show you how it looks like - they have the same logical structure anyways and we shouldn't repeat it. Give me some time :)

Reply:
The only difference of the two buildXXX methods is converting logic a csv line to an user entry - I factor out this it and put them together to one method `buid_dic()`: https://github.com/mocobeta/janome/pull/87/commits/d08383dceac4d4784cd31781bbd9fd02243a26b8